cvmfs/manifest_fetch.cc

```diff
@@ -162,4 +165,31 @@ Failures Fetch(const std::string &base_url, const std::string &repository_name,
   return result;
 }
 
+/**
+ * If the whitelist or the manifest are corrupted, fail-over once to another
+ * stratum 1 if more than a single stratum 1 is known.
+ */
+Failures Fetch(const std::string &base_url, const std::string &repository_name,
+               const uint64_t minimum_timestamp, const shash::Any *base_catalog,
+               signature::SignatureManager *signature_manager,
+               download::DownloadManager *download_manager,
+               ManifestEnsemble *ensemble)
+{
+  Failures result =
+    DoFetch(base_url, repository_name, minimum_timestamp, base_catalog,
+            signature_manager, download_manager, ensemble);
+  if ((result != kFailOk) && (result != kFailLoad) &&
+      (download_manager->num_hosts() > 1))
+  {
+    LogCvmfs(kLogCache, kLogDebug | kLogSyslogWarn,
+             "failed to fetch manifest (%d - %s), trying another stratum 1",
+             result, Code2Ascii(result));
+    download_manager->SwitchHost();
+    result =
+      DoFetch(base_url, repository_name, minimum_timestamp, base_catalog,
+              signature_manager, download_manager, ensemble);
+  }
+  return result;
```

Comment:
I guess we could.  The `DoFetch` routine can do fail-overs internally if hosts are unreachable.  I was afraid of timeouts piling up.  Also, corrupted whitelist/manifest is a pretty rough state for a stratum 1, I'd hope we will never see two at the same time in this state...